.github/workflows/tests.yml

```diff
@@ -93,7 +93,7 @@ jobs:
       run: |
         for file in ${{ steps.files.outputs.all_changed_files }}; do
           if [[ $file == *.py ]]; then
-            if [[ $file != test/* ]]; then
+            if [[ $file != test/* ]] && [[ $file != rest_api/test/* ]]; then
```

Comment:
I think we can get rid of lines 95 and 96 by passing filters to `tj-actions/changed-files`, see https://github.com/tj-actions/changed-files#inputs. Something like this (not tested):
```yaml
    - name: Get changed files
      id: files
      uses: tj-actions/changed-files@v34
      with:
          files:
            - haystack/*
            - *.py
```